classes/mix.bbclass

```diff
@@ -0,0 +1,126 @@
+SYSCONFIG_PREFIX ?= "${sysconfdir}/apps"
+
+FILES_${PN} += "${APP_PREFIX}/${APPNAME}/${APPVERSION}"
+FILES_${PN}-dbg += "${APP_PREFIX}/${APPNAME}/${APPVERSION}/lib/*/priv/obj"
+FILES_${PN}-dbg += "${APP_PREFIX}/${APPNAME}/${APPVERSION}/lib/*/priv/lib/.debug"
+FILES_${PN}-dbg += "${APP_PREFIX}/${APPNAME}/${APPVERSION}/lib/*/priv/.debug"
+FILES_${PN}-dbg += "${APP_PREFIX}/${APPNAME}/${APPVERSION}/erts*/bin/.debug"
+
+FILES_${PN}-staticdev += "${APP_PREFIX}/${APPNAME}/${APPVERSION}/erts*/lib/lib*.a"
+FILES_${PN}-staticdev += "${APP_PREFIX}/${APPNAME}/${APPVERSION}/erts*/lib/internal/lib*.a"
+
+DEPENDS += "avahi erlang-lager-journald-backend elixir-native elixir rebar-native"
+
+python do_download_deps() {
+    import sys
+    import re
+    import urllib2
+    import os
+    import tarfile
+
+    mixlock = open("{dir}/mix.lock".format(dir=d.expand("${S}"))).read()
+    workdir = d.expand("${S}")
+
+    mixlock = mixlock.replace(' ', '').replace('\n', '')[2:-1]
+    mixlock = mixlock.replace('[', '<').replace(']', '>')
+    mixlock = mixlock.replace('{', '[').replace('}', ']')
+    mixlock = mixlock.replace('<', '{').replace('>', '}')
+    mixlock = mixlock.replace(':git', '"git"').replace(':hex', '"hex"')
+    mixlock = mixlock.replace('tag', 'ref')
+    for git_param in ('branch', 'ref'):
+        mixlock = mixlock.replace('%s:'%git_param, '"%s":'%git_param)
+    mixlock = "".join(('{', mixlock, '}'))
+
+    mixlock = re.compile(',:([^,]*),').sub(', "\\1", ', mixlock)
+
+    mixdeps = eval(mixlock)
+
+    if not os.path.exists("{dir}/deps".format(dir=workdir)):
+        os.makedirs("{dir}/deps".format(dir=workdir))
+
+
+    def download_deps(name, uri):
+        bufsize = 16*1024
+        remote = urllib2.urlopen(uri)
+        readbytes = 0
+        file = open("{dir}/deps/{name}".format(dir=workdir,name=name),'wb')
+        while True:
+            data = remote.read(bufsize)
+            if not data: break
+            file.write(data)
+        file.close()
+        tar = tarfile.open("{dir}/deps/{name}".format(dir=workdir,name=name))
+        tar.extractall("{dir}/deps/{name}".format(dir=workdir, name=name.split('-')[0]))
+        tar.close()
+        tar = tarfile.open("{dir}/deps/{name}/contents.tar.gz".format(dir=workdir, name=name.split('-')[0]))
+        tar.extractall("{dir}/deps/{name}".format(dir=workdir, name=name.split('-')[0]))
+        tar.close()
+
+    def clone_debs(name, url, rev):
+        if not os.path.exists("{dir}/deps/{name}".format(dir=workdir, name=name)):
+            os.makedirs("{dir}/deps/{name}".format(dir=workdir, name=name))
+        os.system("git init {dir}/deps/{name}".format(dir=workdir, name=name))
+        os.system("(cd {dir}/deps/{name}; git remote add origin {url})".format(url=url,dir=workdir, name=name))
+        os.system("(cd {dir}/deps/{name}; git fetch origin)".format(dir=workdir, name=name))
+        os.system("(cd {dir}/deps/{name}; git reset --hard {rev})".format(rev=rev,dir=workdir, name=name))
```

Comment:
Please follow libredo/Repotool.py style of accessing a specific git revision. 
Specifically your approach lacks a tracking ref. (use git {init,remote add,fetch, branch,checkout} instead).